agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4944,46 +4944,51 @@ bool slave_thread::handle_channel_preference_query(Socket *sd, ieee1905_1::CmduM
 
     channel_preference_tlv->radio_uid() = network_utils::mac_from_string(config.radio_identifier);
 
-    // Create operating class object
-    auto op_class_channels = channel_preference_tlv->create_operating_classes_list();
-    if (!op_class_channels) {
-        LOG(ERROR) << "create_operating_classes_list() has failed!";
-        return false;
-    }
+    // Get supported channels and operating classes .
+    auto m = get_supported_channels_map();
```

Comment:
Please try to give it a more meaningful name, and define it right before you actually use it.